server/routerlicious/packages/lambdas-driver/src/document-router/documentLambda.ts

```diff
@@ -45,6 +51,19 @@ export class DocumentLambda implements IPartitionLambda {
 			);
 			context.error(error, errorData);
 		});
+		this.contextManager.on(
+			"pause",
+			(lowestOffset: number, pausedAtOffset: number, reason?: any) => {
+				// Emit pause at the lowest offset out of all document partitions
+				// This is important for ensuring that we don't miss any messages
+				// And maintain a range for offsets allowed to be reprocessed on resume, otherwise the contextManager expects offsets only in sequence
```

Comment:
Based on this comment, it sounds like we allow op reprocessing out of order. Is this correct?

Reply:
We should make sure all ops are reprocessed in the correct order, as they could impact the lambda state. One example is in Deli, where we checkpoint for each reprocessed op. If the ops are out of order, we could see an assertion error on the document context because the proposed offset could be behind the tail (last checkpoint).

Reply:
Updated the comment and code to allow moving out of order only once during resume.